source/common/protobuf/message_validator_impl.cc

```diff
@@ -24,12 +24,22 @@ void WarningValidationVisitorImpl::onUnknownField(absl::string_view description)
   if (!it.second) {
     return;
   }
+
   // It's a new field, log and bump stat.
-  ENVOY_LOG(warn, "Unknown field: {}", description);
-  if (counter_ == nullptr) {
-    ++prestats_count_;
+  ENVOY_LOG(warn, "Unexpected field: {}", description);
+  if (unknown_counter_ == nullptr) {
+    ++prestats_unknown_count_;
+  } else {
+    unknown_counter_->inc();
+  }
+}
+
+void WarningValidationVisitorImpl::onDeprecatedField(absl::string_view description,
+                                                     bool soft_deprecation) {
+  if (soft_deprecation) {
+    ENVOY_LOG_MISC(warn, "Unexpected field: {}", absl::StrCat(description, deprecation_error));
   } else {
-    counter_->inc();
+    throw DeprecatedProtoFieldException(absl::StrCat(description, deprecation_error));
   }
```

Comment:
How come the common method is inline now?

Reply:
@htuch if i understand your question correctly, you mean previous version of PR where both visitors shared common method for incrementing stats and logging message? i've changed impl, there is no longer common method. There is little benefit to have common method as deprecated field validation does not increment counters and log messages are different.

Reply:
Yeah, but looking at the two implementations, I'm not seeing a stats increment in either, and still unclear why the warnings are different.